### ISSUE TITLE
Stop clearing artifacts configuration

### ISSUE THREAD

**Issue by Mahoney** [autonomousapps/dependency-analysis-android-gradle-plugin#607]
The following message:
```
Clearing artifacts from the archives configuration. 
 Disable this behavior with -Pdependency.analysis.clear.artifacts=false
 Suppress this warning with -Pdependency.analysis.warnings.silent=true
```
is fairly opaque - I've no idea which artifacts we're talking about, why they were in the archives configuration, whether I do or do not want them there, what the consequence of them being or not being there is, and consequently whether I should disable the behaviour or suppress the warning. I ended up checking out the codebase to discover that this is normal, not a result of something odd I'm doing elsewhere, and that consequently I almost certainly want to suppress the warning.

(Also, suppressing the warning seems to be a generic flag for _all_ warnings - as it happens from examining the codebase I think this is the only one that uses it, but perhaps a more specific flag for this specific case?)

**Comment by autonomousapps** [comment]
I'm open to suggestions here. Maybe this?
```
Clearing artifacts from the archives configuration. If you don't know what this means, don't worry about it.
```

Re
> (Also, suppressing the warning seems to be a generic flag for all warnings - as it happens from examining the codebase I think this is the only one that uses it, but perhaps a more specific flag for this specific case?)

I also considered this, but figured I'd rather cross that bridge when I come to it. I don't want to over-engineer something I may only need once.

**Comment by jjohannes** [comment]
I see the motivation behind the warning, but I think in general it is not "very nice" of a plugin to print so many messages during configuration time for its default setup. That is, I just apply the plugin, do nothing else, and I get warnings (which are not even actionable if I have a good build setup).

This is somehow a blocker right now for me to upgrade (working hard on a build to get rid off all warnings for some time :)).

I would like to see this go to the "info" log only maybe.

I also applied the plugin to my `build-logic`.  And now I have two issues:

**1.**
```
> Task :gradle-build-logic:base-plugins:generatePrecompiledScriptPluginAccessors
Clearing artifacts from the archives configuration.
 Disable this behavior with -Pdependency.analysis.clear.artifacts=false
 Suppress this warning with -Pdependency.analysis.warnings.silent=true
Clearing artifacts from the archives configuration.
 Disable this behavior with -Pdependency.analysis.clear.artifacts=false
 Suppress this warning with -Pdependency.analysis.warnings.silent=true
```
And I think there is no way to avoid it.

**2.**
I need to put `dependency.analysis.warnings.silent=true` into multiple `gradle.properties` files.


Alternatively, it would be nice if "silent warnings" could be configured on an extension instead. But I see how the timing might be a problem with that.


Or maybe the default behavior could be change so that just the "artifacts" that cause trouble are removed from "archives" and then no warning is necessary at all?

**Comment by autonomousapps** [comment]
> Or maybe the default behavior could be change so that just the "artifacts" that cause trouble are removed from "archives" and then no warning is necessary at all?

I actually had a long conversation about this, I think with Louis Jacomet, or maybe Cédric, on the Gradle community slack (a long time ago). We actually tried in a variety of ways to clear specific artifacts, but nothing was working. It has to do with "legacy" behavior of the archives configuration when setting outgoing artifacts for simple inter-project publishing. (One possible solution is improving this story, but I can't prioritize that at this moment.)

The original motivation for clearing the artifacts was this: for some developers, it is part of their development flow to click the "make" button in the IDE, which I believe triggers `assemble`, and this builds everything that is in the archives configuration. As a result, these devs were seeing exploding build times when they were just trying to trigger source generation. With Gradle's encouragement, I simply started clearing that configuration.

I never really felt good about that, because I was silently changing build behavior. And indeed, eventually someone noticed because they had been relying on that old Gradle behavior -- hence the flag to not clear artifacts. 

As I approach the v1 release, I decided I needed to make this behavior more visible so folks could configure it as they see fit. It's a non-ideal situation on top of a non-ideal situation.

I present all this context in case you may have better ideas for handling it. I'm not really comfortable with moving the warning to info, since I don't think many people look at that. Would `quiet` be better? **Alternatively or secondarily, what about ensuring it only gets printed once, instead of once per project?**

Regarding configuring it via extension, you understand correctly that there's a timing problem.

**Comment by jjohannes** [comment]
I don't know if you have tried this, but you should set:

```
isVisible = false
```

for the consumable you create in `publishArtifact(...)`. The name of the flag is misleading - it means that the _Configuration_ is not (so much) "visible" outside the current project. A better name would maybe be "opt-out of legacy behaviour".

So ideally you should always set it to `false` for custom/new configurations unless you need it for some specific legacy behavior. But I also often do not do that, because of the additional _noise_. The flags (and their wrong defaults) are really a PITA. It's all there for backward compatibility, but well.... 

I had a look at the Gradle sources and it looks like this is what you need:
https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/BasePlugin.java#L107

My tests also show that this work and then you can remove this "clearing of artifacts" thingy completely.

**Comment by Vampire** [comment]
Yes, please remove this if possible.
I just spent two hours or so to find out why the hell `assemble` does not depend on `jar` anymore which makes the release process with the `gradle-release` plugin not work as expected anymore.
Turned out to be this plugin silently doing unexpected build modification. (with version 0.80.0)

**Comment by jjohannes** [comment]
I created a PR for the change I proposed: https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/pull/625

**Comment by autonomousapps** [comment]
@Vampire Please do not be rude on this issue tracker. This is your one and only warning.

**Comment by Vampire** [comment]
I don't see where I was rude. I just intended to add my vote and plea to fix this properly with another report of how this had a negative effect on a project. You know that I'm not a native speaker, so if I said something that upsets you, I apology.

**Comment by autonomousapps** [comment]
Given the associated PR, I have updated the issue title to reflect the new goal.
